oeps/oep-0042-arch-authentication.rst

```diff
@@ -0,0 +1,254 @@
+
+OAuth2 and JWTs
+---------------
+
+The currently supported and recommended method of authentication and authorization is an OAuth 2.0 implementation using JSON web tokens (JWTs) as OAuth tokens.
+
+Here are a variety of details around this decision:
+
+* Standardize on the Django OAuth Toolkit (DOT) library to support our OAuth2 implementation.
+
+  * Read the `ADR on Django OAuth Toolkit (DOT)`_.
+
+* JWT Authentication is the standard method of authentication in Open edX.
+
+  * Use JWT and DOT to implement OAuth2. See the `ADR to use JWTs as OAuth Tokens`_ for more details on this decision.
+
+  * JWTs are now the only fully supported type of access token.
+
+  * All JWTs will be signed using "asymmetric" cryptographic keys, as detailed in this `ADR on Asymmetric JWTs`_.
+
+  * For authentication from our micro-frontends, a shared cookie is used to store the JWT and usable across IDAs, as detailed in this `JWT Cookies ADR`_.
+
+Implementation of all the OAuth2/JWT APIs supported by DOT in the LMS Identity Provider can be found in `oauth_dispatch (edx-platform)`_.
+
+* As of the Juniper Open edX release, the deprecated implementation of OAuth2 using OpenID Connect (OIDC) and the Django OAuth Provider (DOP) library has been fully removed. See this `oauth_dispatch as router ADR`_ for additional details of how this transition was implemented in edx-platform, and to better understand the history of ``oauth_dispatch``.
+
+.. _ADR on Django OAuth Toolkit (DOT): https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0002-migrate-to-dot.rst
+.. _ADR to use JWTs as OAuth Tokens: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0003-use-jwt-as-oauth-tokens-remove-openid-connect.rst
+.. _ADR on Asymmetric JWTs: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0008-use-asymmetric-jwts.rst
+.. _JWT Cookies ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0009-jwt-in-session-cookie.rst
+.. _oauth_dispatch (edx-platform): https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/README.rst
+.. _oauth_dispatch as router ADR: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/oauth_dispatch/docs/decisions/0004-oauth-dispatch-as-router.rst#L33
+
+
+OAuth2 and Bearer Tokens
+------------------------
+
+This section refers to Bearer Tokens as documented in `OAuth 2.0 RFC: Bearer Token Usage`_.
+
+Currently, only the mobile applications (iOS and Android) should continue to use Bearer Tokens. In this method, the LMS issues a token to the mobile apps and the mobile apps include this token in their header when hitting IDA endpoints. The IDAs verify the token and either service or reject the request, based on the token validity.
```

Comment:
`should continue` -> `continue`

Eventually, we like for the mobile apps to also convert to using JWTs instead.

Reply:
I think we should clarify language around "bearer token". Our JWTs are one kind of bearer token in the sense of the OAuth 2 spec. If there's a different kind of bearer token that we were using previously, it would be good to have a name for it that can be distinguished from the more general name.

Reply:
@timmc-edx: I tried to add clarity.  Let me know.